src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMEnvironmentContributor.java

```diff
@@ -0,0 +1,89 @@
+package com.atlassian.bitbucket.jenkins.internal.scm;
+
+import com.google.common.annotations.VisibleForTesting;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.model.*;
+import hudson.plugins.git.GitSCM;
+import hudson.scm.SCM;
+import jenkins.branch.BranchSource;
+import jenkins.branch.MultiBranchProject;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+
+import java.io.IOException;
+import java.util.Objects;
+
+@Extension
+public class BitbucketSCMEnvironmentContributor extends EnvironmentContributor {
+
+    public void buildEnvironmentFor(@NonNull Job job, @NonNull EnvVars envs, @NonNull TaskListener listener) throws IOException, InterruptedException {
+        if (job instanceof AbstractProject) {
+            // Case 1 - Freestyle job
+            SCM scm = ((AbstractProject<?, ?>) job).getScm();
+            if (!(scm instanceof BitbucketSCM)) {
+                return;
+            }
+            ((BitbucketSCM) scm).getBitbucketSCMRepository().buildEnvironment(envs);
+            return;
+        }
+        if (job instanceof WorkflowJob) {
+            // Case 2 - Not a freestyle job. Proceed to inspect SCM on item
+            WorkflowJob workflowJob = (WorkflowJob) job;
+            SCM scm = workflowJob.getTypicalSCM();
```

Comment:
This does not do what you want it to do. 
If you have two different types of SCMs (say a git and a Bitbucket one) this will always return `null`. I think you need to get all SCMs out and filter them.